### PR TITLE
Implemented crdcdh-903

### PR DESCRIPTION
**Pull request by vshand11**
1) add find last batch by fileName function in mongo_dao.py 
2) updated file_validator.py to rewrite the extra file error based on requirements in the ticket.
3) update data_loader.py to convert nan to None in dataframe to avoid issues in metadata validation.
4) add node type validation in essential_validator.py

### CODE REVIEW

File: src/file_validator.py [hunk 1]
```diff
@@ -321,11 +321,24 @@ def validate_all_files(self, submission_id):
...
                     break
                 file_name = file.key.split('/')[-1]
-                
+               
                 if file_name not in manifest_file_names:
-                    msg = f'File “{file_name}”: no record found.'
+                    file_batch = self.mongo_dao.find_batch_by_file_name(submission_id, "data file", file_name)
+                    batchID = file_batch[ID] if file_batch else "-"
+                    displayID = file_batch["displayID"] if file_batch else "-"
+                    msg = f'Extra data file “{file_name}” is found in CRDC datahub storage.'
```
Comment (n2iw) [inline src/file_validator.py:329]: According to the error message document, this should be "**File “{file_name}”: no record found**"

File: src/file_validator.py [hunk 2]
```diff
@@ -321,11 +321,24 @@ def validate_all_files(self, submission_id):
...
                     self.log.error(msg)
-                    error = create_error("Extra file found", msg)
+                    error = {
+                        TYPE: "Data File",
+                        "validationType": "file",
```
Comment (n2iw) [inline src/file_validator.py:333]: Please change it to "data file", I'll update the ticket also.

**Review by n2iw — CHANGES_REQUESTED** [review]

File: src/file_validator.py (hunk 1 above)
Comment (vshand11) [reply src/file_validator.py]: Updated.

File: src/file_validator.py (hunk 2 above)
Comment (vshand11) [reply src/file_validator.py]: updated.

File: src/data_loader.py [hunk 3]
```diff
@@ -53,6 +54,7 @@ def load_data(self, file_path_list):
             try:
                 df = pd.read_csv(file, sep=SEPARATOR_CHAR, header=0, encoding=UTF8_ENCODE)
                 df = (df.rename(columns=lambda x: x.strip())).apply(lambda x: x.str.strip() if x.dtype == 'object' else x) # stripe white space.
+                df = df.replace({np.nan: None})  # replace Nan in dataframe with None
```
Comment (n2iw) [inline src/data_loader.py:57]: Please remove this and fix the nan issue in a separate PR, thanks!

**Review by n2iw — CHANGES_REQUESTED** [review]

**Review by n2iw — APPROVED** [review]
LGTM

**Pull request merged**